### ISSUE TITLE
[Website] Remove mentions of master branch from Apache Arrow website content

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#20161]
This is a follow up task to [ARROW-15694](https://issues.apache.org/jira/browse/ARROW-15694) and [ARROW-15988](https://issues.apache.org/jira/browse/ARROW-15988).

We need to update the actual content of the Apache Arrow website to no longer refer to the "master" branch. This includes updating links, text snippets, and the `README.md` for `{}arrow-site{`}. There may also be other files that need to be updated, as well.

**Reporter**: [Kevin Gurney](https://issues.apache.org/jira/browse/ARROW-15991) / @kevingurney

<sub>**Note**: *This issue was originally created as [ARROW-15991](https://issues.apache.org/jira/browse/ARROW-15991). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by kevingurney** [comment]
I've done some preliminary investigation of this issue.

It seems like the uses of "master" in `apache/arrow-site` can be categorized as follows:

1. Uses in the `_release` directory. It's unclear whether it makes sense to change these uses since these are referring to a past state of the repository.
2. Uses in the `_posts` directory. Similar to 1., it's not clear whether it makes sense to change these uses.
3. Uses in `index.html`. All these uses appear to be links referring to files/directories in the `apache/arrow` directory. Therefore, we should be able to change this to `main` now that the default branch of the `apache/arrow` repo has been renamed to `main`. 
4. Uses in `overview.md`. Same as 3.
5. Uses in `_includes/header.html`. Same as 3.
6. Uses in `README.md`. It seems like we could change the use of "master" in the phrase "On commits to the `master` branch of `apache/arrow-site`" to "On commits to the default branch of `apache/arrow-site`". Alternatively, we could change "master" to "main" after the default branch has been renamed to "main" for the `apache/arrow-site` repo.
7. Uses in `install.md`. This is referring to `https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-arrow`. We could change this to `https://github.com/msys2/MINGW-packages/tree/HEAD/mingw-w64-arrow`.
8. Uses in `powered_by.md`. The only use is in `https://github.com/apache/arrow-site/edit/master/powered_by.md`. It might be OK to change this to `https://github.com/apache/arrow-site/edit/HEAD/powered_by.md`. Although, it is slightly unclear to me whether referring to `HEAD` here is actually equivalent. Otherwise, changing this to "main" is dependent on the `apache/arrow-site` default branch being renamed to "main" first.
9. Uses in `community.md`. The only use is in the phrase "commits@ for commits to the [apache/arrow](https://github.com/apache/arrow) and [apache/arrow-site](https://github.com/apache/arrow-site) repositories (typically to master only)...". Changing this use depends on the default branch of `apache/arrow-site` being renamed to "main".
10. Uses in `.github/workflows/deploy.yml`. Removing these uses depends on the default branch of `apache/arrow-site` being renamed to `main`. See #31412, which is being used to track this usage.

---

Since removing several of the uses of "master" in the Apache Arrow website are dependent on the default branch name of the `apache/arrow-site` repo being renamed to "main", it might make sense to consider this issue "non-blocking" for the time being and focus on getting the default branch renamed to "main" for `apache/arrow-site`. Once the rename is complete, I can open a PR to remove the remaining uses of "master".

Alternatively, I am also more than happy to create a PR now to address the uses above which aren't dependent on the default branch of `apache/arrow-site` being renamed to `main` and to follow up with another PR to remove the remaining uses after the default branch rename is complete. I can follow whichever approach the community feels makes the most sense.

**Comment by kevingurney** [comment]
take

**Comment by kevingurney** [comment]
I've opened PR apache/arrow-site#324 to address this issue.
